tests/api_tests/test_site_hierarchy_api.py

```diff
@@ -0,0 +1,178 @@
+from rdr_service.model.hpo import HPO
+from rdr_service.model.organization import Organization
+from rdr_service.model.site import Site
+from rdr_service.model.site_enums import ObsoleteStatus
+from rdr_service.participant_enums import OrganizationType
+
+
+def _make_expected_response(**kwargs):
+    full_site_hierarchy = {
+        "data": [
+            {
+                "awardee_id": "PITT",
+                "organization_id": "PITT_BANNER_HEALTH",
+                "site_name": "Monroeville Urgent Care Center",
+                "google_group": "hpo-site-monroeville",
+            },
+            {
+                "awardee_id": "PITT",
+                "organization_id": "PITT_BANNER_HEALTH",
+                "site_name": "Phoenix Urgent Care Center",
+                "google_group": "hpo-site-bannerphoenix",
+            },
+            {
+                "awardee_id": "AZ_TUCSON",
+                "organization_id": "AZ_TUCSON_BANNER_HEALTH",
+                "site_name": "Phoenix clinic",
+                "google_group": "hpo-site-clinic-phoenix",
+            },
+            {
+                "awardee_id": "ILLINOIS",
+                "organization_id": "ILLINOIS_1",
+                "site_name": "Illinois Site 1",
+                "google_group": "illinois-site-1",
+            },
+        ]
+    }
+
+    filtered_data = [
+        item
+        for item in full_site_hierarchy["data"]
+        if all(item[key] == value for key, value in kwargs.items())
+    ]
+
+    return {"data": filtered_data} if kwargs else full_site_hierarchy
+
+
+class SiteHierarchyApiTest(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.hpo_dao = HPODao()
+        self.org_dao = OrganizationDao()
+        self.site_dao = SiteDao()
+        self.hpo_dao.insert(
+            HPO(
+                hpoId=ILLINOIS_HPO_ID,
+                name="ILLINOIS",
+                displayName="illinois",
+                organizationType=OrganizationType.HPO,
+            )
+        )
+        self.org_dao.insert(
+            Organization(
+                organizationId=ILLINOIS_ORG_ID,
+                externalId="ILLINOIS_1",
+                displayName="Illinois 1",
+                hpoId=ILLINOIS_HPO_ID,
+            )
+        )
+        self.site_dao.insert(
+            Site(
+                siteId=ILLINOIS_SITE_ID,
+                siteName="Illinois Site 1",
+                googleGroup="illinois-site-1",
+                organizationId=ILLINOIS_ORG_ID,
+                hpoId=ILLINOIS_HPO_ID,
+                isObsolete=ObsoleteStatus.ACTIVE,
+            )
+        )
+        self.site_dao.insert(
+            Site(
+                siteId=OBSOLETE_ID,
+                siteName="Illinois Site 2",
+                googleGroup="illinois-site-2",
+                organizationId=ILLINOIS_ORG_ID,
+                hpoId=ILLINOIS_HPO_ID,
+                isObsolete=ObsoleteStatus.OBSOLETE,
+            )
+        )
```

Comment:
So in the `BaseTestCase` we have a instantiation to the `DataGenerator` class which we use for data generation mainly in the unit tests. There is a method for site that you can utilize in that class that will help clean some of these unneeded instance variables.

Reply:
Done!